source/extensions/filters/common/lua/lua.h

```diff
@@ -60,6 +60,28 @@ namespace Lua {
  */
 #define DECLARE_LUA_CLOSURE(Class, Name) DECLARE_LUA_FUNCTION_EX(Class, Name, lua_upvalueindex(1))
 
+/**
+ * Calculate the maximum space needed to be aligned.
+ */
+template <typename T> constexpr size_t maximumSpaceNeededToAlign() {
+  return alignof(T) > alignof(void*) ? sizeof(T) : sizeof(T) + alignof(T) - 1;
+}
+
+/**
+ * Create a new user data and assign its metatable.
+ */
+template <typename T> inline void* allocateLuaUserData(lua_State* state) {
+  auto space = maximumSpaceNeededToAlign<T>();
+  void* mem = lua_newuserdata(state, space);
+  luaL_getmetatable(state, typeid(T).name());
+  ASSERT(lua_istable(state, -1));
+  lua_setmetatable(state, -2);
+
+  // Align the allocated memory.
+  // TODO(dio): should we do #ifdef for skipping aligning the memory?
+  return std::align(std::alignment_of<T>(), sizeof(mem), mem, space);
```

Comment:
Is this right? From reading the docs I think it should be `sizeof(T)` and not `sizeof(mem)` ?